src/coreComponents/dataRepository/wrapperHelpers.hpp

```diff
@@ -627,20 +668,29 @@ std::unique_ptr< int > averageOverSecondDim( T const & )
   return std::unique_ptr< int >( nullptr );
 }
 
-template< typename T >
-typename std::enable_if_t< traits::is_array_type< T >, int >
-numArrayDims( T const & GEOSX_UNUSED_PARAM( var ) )
+template< typename T, int NDIM, int USD >
+int numArrayDims( ArrayView< T const, NDIM, USD > const & GEOSX_UNUSED_PARAM( var ) )
```

Comment:
Can't we just use `var.NDIM` or put a member function in `ArrayView` to give access to NDIM?

Reply:
That's what the previous version did, returning `T::NDIM`. I think this way is cleaner than using `traits::is_array_type` and `enable_if`.